Assets/HoloToolkit/SpatialUnderstanding/Scripts/SpatialUnderstandingCustomMesh.cs

```diff
@@ -0,0 +1,182 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using UnityEngine;
+using System.Collections;
+using System;
+using System.Runtime.InteropServices;
+using System.Collections.Generic;
+using UnityEngine.VR.WSA;
+
+namespace HoloToolkit.Unity
+{
+    /// <summary>
+    /// Handles the custom meshes generated by the understanding dll. The meshes
+    /// are generated during the scanning phase and once more on scan finalization.
+    /// The meshes can be used to visualize the scanning progress.
+    /// </summary>
+    public class SpatialUnderstandingCustomMesh : SpatialMappingSource
+    {
+        // Config
+        [Tooltip("Indicate the time between mesh imports, during the scanning phase. A value of zero will disable pulling meshes from the dll")]
+        public float ImportMeshPeriod = 1.0f;
+        [Tooltip("Material used to render the custom mesh generated by the dll")]
+        public Material MeshMaterial;
+
+        // Properties
+        /// <summary>
+        /// Controls rendering of the mesh. This can be set by the user to hide or show the mesh.
+        /// </summary>
+        public bool DrawProcessedMesh
```

Comment:
There is an open request to change 'DrawVisualMeshes' to 'DrawSurfaces', so I would not try matching the SpatialMappingManager. Also, since this flag controls rendering of the 'processed' meshes as opposed to the 'raw' meshes, the name is more accurate. I would probably leave it singular here, since it matches the class name.